apps/sms/views/conversation/js/conversation.js

```diff
@@ -1519,13 +1530,13 @@ var ConversationView = {
     }).bind(this);
 
     var onRenderMessage = (function renderMessage(message) {
-      if (this._stopRenderingNextStep) {
+      if (!this.shouldRenderForThreadId(threadId)) {
         // stop the iteration and clear the taskQueue
         taskQueue = null;
         return false;
       }
       taskQueue.push(() => {
-        if (!this._stopRenderingNextStep) {
+        if (this.shouldRenderForThreadId(threadId)) {
```

Comment:
maybe someday our TaskQueue can have ```clear/cancel``` method that will automagically not allow to execute next step and return rejected promise, however there likely be overhead for the wrapper function that will perform this check :)